pincer/client.py

```diff
@@ -581,6 +586,63 @@ async def payload_event_handler(self, _, payload: GatewayDispatch):
         """
         await self.process_event("payload", payload)
 
+    async def wait_for(
+        self,
+        event_name: str,
+        check: CheckFunction = None,
+        timeout: Optional[float] = None
+    ):
+        """
+        Parameters
+        ----------
+        event_name : str
+            The type of event. It should start with `on_`. This is the same
+            name that is used for @Client.event.
+        check : Callable[[Any], bool], default=None
+            This function only returns a value if this return true.
+        timeout: Union[float, None]
+            Amount of seconds before timeout.
```

Comment:
Should replace the old typehints
```suggestion
        check : CheckFunction
            This function only returns a value if this return true.
        timeout: Optional[float]
            Amount of seconds before timeout.
```